### PR TITLE
changes

### PR DESCRIPTION
**Pull request by DonglinLiJCU**
@xm-y02

### CODE REVIEW

File: prac5/state_names.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    state_code = input("Enter short state: ")
+
+    # if str.upper(state_code) in CODE_TO_NAME:
+    #     print(state_code, "is", CODE_TO_NAME[str.upper(state_code)])
+    # else:
+    #     print("Invalid short state")
+    # state_code = input("Enter short state: ")
+
+for item in CODE_TO_NAME.items():
+    print(f"{item[0]:<3} is {item[1]}")
```
Comment (xm-y02) [inline prac5/state_names.py:26]: output didn't neatly line up

File: prac5/hex_colours.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+CODE_TO_HEX_COLOR = {"ABSOLUTE ZERO": "#0048ba", "ACID GREEN": "#b0bf1a", "ALICEBLUE": "#f0f8ff",
+                     "ALIZARIN CRIMSON": "#e32636",
+                     "AMARANTH": "#e52b50", "AMBER": "#ffbf00", "AMETHYST": "#9966cc", "ANTIQUEWHITE": "#faebd7",
+                     "ANTIQUEWHITE1": "#ffefdb"}
+
+color_name = input("Enter color name: ")
```
Comment (xm-y02) [inline prac5/hex_colours.py:6]: I think add a ".lower" here can be better

File: prac5/hex_colours.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+while color_name != "":
+    try:
+        print(color_name, "is", CODE_TO_HEX_COLOR[str.upper(color_name)])
+    except KeyError:
+        print("Invalid color name")
```
Comment (xm-y02) [inline prac5/hex_colours.py:11]: problems said "should not crash the program" so maybe don't print invalid message...?

File: prac5/emails.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+"""
+Emails
+Estimate: 10 minutes
+Actual:   15 minutes
+"""
+
+email_name = {}
```
Comment (xm-y02) [inline prac5/emails.py:7]: I think put dictionary inside the main will be better

File: prac5/emails.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def main():
+    email = input("Email: ")
+    while email != "":
+        choice = input(f"Is your name {analyse_name(email)}? (Y/n) ")
```
Comment (xm-y02) [inline prac5/emails.py:13]: "get name" part could be written as an function

File: prac5/hex_colours.py (hunk 2 above)
Comment (DonglinLiJCU) [reply prac5/hex_colours.py:6]: There is no need to add lower

**Pull request merged**